boards/BMS/Src/App/states/App_AllStates.c

```diff
@@ -124,10 +124,13 @@ bool App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
     struct Charger *         charger     = App_BmsWorld_GetCharger(world);
 
     const bool charger_is_connected = App_Charger_IsConnected(charger);
-    bool       status               = true;
+    const bool balancing_enabled    = App_CanRx_Debug_CellBalancing_RequestCellBalancing_Get();
+    const bool ignore_other_boards  = charger_is_connected || balancing_enabled;
+
+    bool status = true;
 
     // ignore heartbeat when charging, other boards likely disconnected
-    bool missing_hb = charger_is_connected ? false : App_SendAndReceiveHeartbeat(hb_monitor);
+    bool missing_hb = ignore_other_boards ? false : App_SendAndReceiveHeartbeat(hb_monitor);
```

Comment:
nit - this may be simpler
```suggestion
    bool missing_hb = ignore_other_boards & App_SendAndReceiveHeartbeat(hb_monitor);
```